src/templates/_select.twig

```diff
@@ -21,114 +22,123 @@
+        add_date_class(commerce_wrapper)
+
+        const commerce_input = commerce_wrapper.querySelector(
+          'input[class="{{ filter_class }}_{{ now }}"]'
+        )
+
+        if (commerce_input) {
+          add_date_class(commerce_input)
+          const commerce_select = commerce_wrapper.querySelector(
+            'select[class="{{ name }}_{{ now }}"]'
+          )
+
+          if (commerce_select) {
+            add_date_class(commerce_select)
+
+            const commerce_select_array = [...commerce_select.options]
+
+            commerce_input.addEventListener('input', (event) => {
+              if (event.target.value) {
+                commerce_select_array.forEach((option) => {
+                  const regex = new RegExp(event.target.value, 'ig')
+                  if (regex.test(option.textContent)) {
+                    option.disabled = false
+                    option.hidden = false
+                  } else {
+                    option.disabled = true
+                    option.hidden = true
+                  }
+                })
+              } else {
+                commerce_select_array.forEach((option) => {
+                  option.disabled = false
+                  option.hidden = false
+                })
+              }
+            })
+
+            const commerce_selected_only = commerce_wrapper.querySelector(
+              'input[class="{{ selected_only_class }}_{{ now }}"]'
+            )
+            add_date_class(commerce_selected_only)
+
+            commerce_selected_only.addEventListener('change', (event) => {
+              if (event.target.checked) {
+                commerce_select_array.forEach((option) => {
+                  if (option.selected) {
+                    option.disabled = false
+                    option.hidden = false
+                  } else {
+                    option.disabled = true
+                    option.hidden = true
+                  }
+                })
+              } else {
+                commerce_select_array.forEach((option) => {
+                  option.disabled = false
+                  option.hidden = false
+                })
+              }
+            })
+
+            const commerce_clear_selected = commerce_wrapper.querySelector(
+              'button[class="{{ clear_selected_class }}_{{ now }}"]'
+            )
+            add_date_class(commerce_clear_selected)
+
+            commerce_clear_selected.addEventListener('click', (event) => {
+              event.preventDefault()
+              commerce_select_array.forEach((option) => {
+                if (option.selected) {
+                  option.selected = false
                 }
-            }
+              })
+            })
+          }
         }
-    };
+      }
+    }
 
-    attach_events();
+    attach_events()
     document.addEventListener('DOMContentLoaded', () => {
-        attach_events();
-    });
-})();
+      attach_events()
+    })
+  })()
```

Comment:
Can we keep this at 4 spaces with semicolons? I know we use StandardJS normally, but this repo doesn't follow that convention and neither does our Craft code.

Reply:
Yeah, I'll do that.